modules/sentinel-watchlist/main.tf

```
@@ -18,3 +18,9 @@ resource "azurerm_sentinel_watchlist" "watchlist" {
   default_duration           = var.default_duration
   labels                     = var.labels
 }
+
+resource "azurerm_sentinel_watchlist_item" "watchlist_item" {
+  for_each     = { for index, elem in var.properties : index => elem }
```

Index isn't needed. So instead of creating a map, we should create a list.

```suggestion
  for_each     = var.properties
```

---

var.properties is a list of a map(string)
the for_loop is iterating through it and converting it to a map(string)

---

Ah, i misunderstood. Ok, let me commit that.